### ISSUE TITLE
SVM optimizer

### ISSUE THREAD

**Issue by jaberg** [hyperopt/hyperopt#121]
http://www.keerthis.com/report_hypergrad.pdf

(This is a longer version of a 2006 NIPS paper)

Write a specific SVM hyper-parameter optimizer for e.g. sklearn's SVM algorithms. This part would implement the math from that tech report (Sections 4 and 5) and compute the gradients for particular SVM hyper-parameters.

This requires the gradient-based search protocol described in #120
